src/signTx.c

```diff
@@ -1395,11 +1393,27 @@ static void signTx_handleWithdrawal_ui_runStep()
 		ui_displayAdaAmountScreen("Withdrawing rewards", txBodyCtx->stageData.withdrawal.amount, this_fn);
 	}
 	UI_STEP(HANDLE_WITHDRAWAL_STEP_DISPLAY_PATH) {
-		reward_account_t rewardAccount = {
-			.keyReferenceType = KEY_REFERENCE_PATH,
-			.path = txBodyCtx->stageData.withdrawal.path
-		};
-		ui_displayRewardAccountScreen(&rewardAccount, ctx->commonTxData.networkId, this_fn);
+		if (STAKE_CREDENTIAL_KEY_PATH == txBodyCtx->stageData.withdrawal.stakeCredential.type) {
+			reward_account_t rewardAccount = {
+				.keyReferenceType = KEY_REFERENCE_PATH,
+				.path = txBodyCtx->stageData.withdrawal.stakeCredential.pathSpec
+			};
+			ui_displayRewardAccountScreen(&rewardAccount, ctx->commonTxData.networkId, this_fn);
+		} else {
+			reward_account_t rewardAccount = {
+				.keyReferenceType = KEY_REFERENCE_HASH,
+			};
+			constructRewardAddressFromHash(
+				ctx->commonTxData.networkId,
+				REWARD_HASH_SOURCE_SCRIPT,
+				txBodyCtx->stageData.withdrawal.stakeCredential.scriptHash,
+				SIZEOF(txBodyCtx->stageData.withdrawal.stakeCredential.scriptHash),
+				rewardAccount.hashBuffer,
+				SIZEOF(rewardAccount.hashBuffer)
+			);
+			ui_displayRewardAccountScreen(&rewardAccount, ctx->commonTxData.networkId, this_fn);
```

Comment:
Ideally, we want to end the `UI_STEP` with a single call to `ui_display...`, not having a ui call in every if branch.

Reply:
Done, though it makes the variable live in a larger scope. This is the one return or multiple returns consideration just a different colour of it.

Reply:
Only partially --- `return` ends the function while `ui_` does not.